### DISCUSSION TITLE
kakfa scaler - the scaled consumer pod is not fixed

### DISCUSSION THREAD

**Discussion by mengxu2018** [kedacore/keda#1734] [Q&A / Need Help]
My requirement is to scale the consumer service with two topics.
I use one scaledobject with two triggers to do it.
the topic 1 and topic 1 both have three partitions.
When I emit 1000 messages to both the two topics, and did four times of tests.
The maxReplicas is not fixed, it might be 3, 4 or 5. Could anyone give me any explanation about it?
![image](https://user-images.githubusercontent.com/20497420/113985395-aff06280-987e-11eb-8c08-81cc30dd4ae3.png)
Above is the screenshot of my 4 round test.
You can see the replicas will change from 0-1-3-4-0, or 0-1-4-5-3-0, or 0-1-3-0.
especially when the targets is 
![image](https://user-images.githubusercontent.com/20497420/113985850-2d1bd780-987f-11eb-84c4-7b12864febf6.png)
it will change to 3 or 4 replicas in next phase.

Thanks.

**Comment by zroubalik** [comment]
Could you please past here your ScaledObject?

And pardon me, I am not sure I fully understand your question. What exactly ia your problem?

**Reply by mengxu2018** [reply]
thanks for you response, I re-edit the post.
You can see the column of screenshot I marked red.
I did four round test(send 1000 messages to both of the two topics)
but the scaled number will be 3 or 4 or 5.  Could you give me any explanations?
Does it use the https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#algorithm-details.?

**Reply by zroubalik** [reply]
I don't see the re-edited post. Yes it is using that algorithm, if you use more triggers it will use the max value as a target.

And what scale number do you expect it to be?
